File: apps/notification/src/model/Device.ts
'use strict'

class Device {
    userId: string;
    token: string;
    createDate: string;
    createBy: string;
    updateDate: string;
    updateBy: string;
    constructor(body: object) {
        // @ts-ignore
        const {userId, token, createDate, updateDate, updateBy, createBy} = body;

        // data fields
        this.userId = userId
        this.token = token

        // audit fields
        this.createDate = createDate
        this.createBy = createBy
        this.updateDate = updateDate
        this.updateBy = updateBy
    }
}

export {Device}

File: apps/notification/src/model/device.py
from dataclasses import dataclass
from typing import Optional

from src.dao.device_dao import DeviceDocument


@dataclass
class Device:
    user_id: str
    token: str
    create_date: str
    update_date: str

    @staticmethod
    def from_document(document: Optional[DeviceDocument]):
        if not document:
            return None

        return Device(
            user_id=document.get("_id"),
            token=document.get("token"),
            create_date=document.get("createDate"),
            update_date=document.get("updateDate")
        )
